Sources/Client/Client/Client.swift

```diff
@@ -95,6 +95,7 @@ public final class Client {
     
     /// Weak references to channels by cid.
     let watchingChannelsAtomic = Atomic<[ChannelId: [WeakRef<Channel>]]>([:])
+    let rewatchingChannelIdsAtomic = Atomic<Set<ChannelId>>([])
```

Comment:
Why do we need this as another variable? Can't we just use
```swift
watchingChannelsAtomic.keys
```

Reply:
I think you missed the point of usage. It's for storing channel ids that were watched.

Reply:
Isn't it the same with `watchingChannelsAtomic`? Missing some context here probably 😓

Reply:
`watchingChannelsAtomic` is for keeping channels synchronized with unread count and watcher count. All channels from queries will be added to it.
`rewatchingChannelIdsAtomic` is for channels that were watched explicitly. It's important to distinguish them because for query channels with watch flag will affect a lot of channels and after reconnection the recommended flow to re-request query for channels with all changes in messages and unread counts. So if we will rewatch all channels we will get query channels with watch flag and massive watch requests for each channel. It's just not necessary.

Reply:
I see, so after reconnection unreadCount might be wrong for some channels because they were not watched explicitly?

Reply:
It's not about that.
It's all for LLC.
We have a _bug_ if you start watch a channel it will stop we reconnection (e.g. Internet connection toggled).
This PR fixes that.

Reply:
No worries about the unread count for channels. All of them from `watchingChannelsAtomic` will be updated correctly.